pywemo/ssdp.py

```diff
@@ -345,39 +361,29 @@ def scan(  # noqa: C901
                 entry = UPNPEntry.from_response(response)
                 if entry.usn == VIRTUAL_DEVICE_USN:
                     continue  # Don't return the virtual device.
-                if entry.description is not None:
-                    device = entry.description.get('device', {})
-                    mac = device.get('macAddress')
-                    serial = device.get('serialNumber')
-                    services = device.get("serviceList", {}).get("service", [])
-                    service_types = [
-                        service.get("serviceType")
-                        for service in services
-                        if isinstance(service, dict)
-                    ]
-                else:
-                    mac = None
-                    serial = None
-                    service_types = []
 
                 # Search for devices
-                if any(i is not None for i in [st, match_mac, match_serial]):
-                    if not entry_in_entries(entry, entries, mac, serial):
-                        if match_mac is not None:
-                            if match_mac == mac:
-                                entries.append(entry)
-                        elif match_serial is not None:
```

Comment:
It wasn't clear to me why `elif` is used here. Was that intentional? What should the behavior be if both `match_mac` and `match_serial` are provided?

Reply:
Taking a step back: Could we remove match_mac, match_serial & match_st and just keep match_udn? That would clean-up the code a bit and remove the question above.

I don't have the context on why match_mac & match_serial were added. I also don't have the context for why `mac` is a required argument to `Device.__init__`. It seems odd to me that discovery parses /setup.xml to extract the mac address only to pass it to `Device.__init__` which then parses /setup.xml again.

Reply:
I've had some of the same questions myself, so maybe @pavoni recalls some of the history here?

Reply:
Some of this is before my time (even), but at some point the discovery was integrated with HA’s discovery, which I think passed the mac.

So may be ghosts of how HA worked long ago.

As long as it works stand along and inside HA, fee free to refactor.

Rediscovery has always been shaky, so would be great to fix.

Reply:
Thanks!

I've made `mac` optional and added a deprecation warning so that we can remove it in the future. I also removed `match_mac` & `match_serial`.

This greatly speeds up `ssdp.scan` since it no longer needs to read setup.xml. All time in `ssdp.scan` is spent ... on scanning. :)

And, since setup.xml no longer needs to be read as part of `ssdp.scan` I've also removed the `DESCRIPTION_CACHE`. IIRC it was only there as an optimization since it's typical that `ssdp.scan` receives multiple responses from the same device.